### PR TITLE
Create edit and approve page component

### LINKED ISSUE #70: Create edit and approve page component
Create an Edit & Approve page component matching the screenshot below. The header in the screenshot should be ignored (and will be completed in a separate issue). Also note the screenshot is incorrect - the white page background should extend the full height.

The contents of this page should fill 100% of the available width. There is a separate issue to create the wrapper that will constrain the width and center the content on the page. For this issue, it is expected that the content will span the full width of the browser (and will later be width-constrained when integrated with the wrapper component).

Please follow the instructions in [OUTREACHY.md](https://github.com/Pocket/curation-tools-frontend/blob/main/OUTREACHY.md) when developing the component.

All tasks should be completed using Typescript.

edit-and-approve

### LINKED ISSUE #93: ensure all components use the custom button component
use our custom button component instead of the material UI button component.

### PR DESCRIPTION
**Pull request by nina-py**
## Implementation details

- Added EditAndApproveStory component with all the required fields.

- Deleted unneeded .css files.

- Added a global theme, making solid green and brick red primary
and secondary colors respectively.

- Global theme contains style overrides for all the buttons.

- Updated all components that use buttons to use the updated button
code.

- Added styles for the switch component.

- Fixed minor layout issues in several components to closer match the screenshots in Figma for each component.

- Props are passed into the form (uncontrolled for now).

- Tests have been added for the form. 

Closes #70. Closes #93.

## Buttons

Quite a bit of work went into theming the buttons and updating components to use the new buttons as each component styled the buttons in its own way thing prior to this change. I have had to update the layout of some of these components (i.e. the login form) as well to match the screenshots supplied in completed issues. 

At the moment, `App.tsx` contains all four components affected so that the changes can easily be reviewed. These are the new `EditAndApproveStory` component, `AddStory`, `Card`, and `LoginForm`. The `Header` component also uses the new buttons but more work is needed there - the way it's set up right now it's not taking advantage of Material UI's responsive styles. This should probably be addressed in a separate PR.

This is how buttons can be styled in the app now with the custom Button component:

```jsx
// Positive (default), i.e. "Approve":
<Button>Approve</Button>
// or, if explicitly defining the type:
<Button buttonType="positive">Approve</Button>

// Negative, i.e. "Reject":
<Button buttonType="negative">Reject</Button>

// Neutral, i.e. "Snooze":
<Button buttonType="neutral">Snooze</Button>

// Hollow, i.e. "Log Out":
<Button buttonType="hollow">Log Out</Button>

//Neutral & hollow, i.e. "Cancel":
<Button buttonType="hollow-neutral">Cancel</Button>
```

### CODE REVIEW

**Comment by nina-py** [comment]
For the switch component, I have adapted styles from an AirBnB-inspired switch here: https://mui-treasury.com/styles/switch/ (GitHub: https://github.com/siriwatknp/mui-treasury). It's mostly fine but it is off by one (!) pixel in the "on" position. I will hunt this down later; for now I think it's good to go as is. 

Just the props/tests remaining, not much left to do in this PR.

**Comment by nina-py** [comment]
Just a note that tests are passing but output is flooded with warnings identical to the ones mentioned here: https://github.com/testing-library/dom-testing-library/issues/774. The suggested fix doesn't fix the warnings for me but I will look into it in more detail over the coming days.

**Comment by nina-py** [comment]
All issues with the tests and the Switch component styles have been fixed. Please review - I think this is good to go. All affected components (Card, AddStory, EditAndApproveStory, and LoginForm) are rendered on the home page temporarily. Header is also using the updated buttons but needs refactoring to use MUI Grid components, so please forgive the misalignment of some elements for now.

File: src/components/EditAndApproveStory/EditAndApproveStory.tsx [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+   */
+  author: string;
+  /**
+   * The URL of the story
+   */
+  url: string;
+  /**
+   * The title of the story
+   */
+  headline: string;
```
Comment (kkyeboah) [inline src/components/EditAndApproveStory/EditAndApproveStory.tsx]: Update to `title`
```suggestion
  title: string;
```

File: src/components/EditAndApproveStory/EditAndApproveStory.tsx [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+
+  /**
+   * The topic of the story, i.e. "History"
+   */
+  topic: string;
+
+  /**
+   * Curation history of this story
+   */
+  history: string[];
```
Comment (kkyeboah) [inline src/components/EditAndApproveStory/EditAndApproveStory.tsx]: remove `history`
```suggestion
```

File: src/components/Card/Card.tsx [hunk 3]
```diff
@@ -44,56 +48,50 @@ const useStyles = makeStyles(() =>
 );
 
 interface CardProps {
-  imageUrl: string;
-  title: string;
+  thumbnailUrl: string;
+  headline: string;
```
Comment (kkyeboah) [inline src/components/Card/Card.tsx]:
```suggestion
  title: string;
```

File: src/components/CardText/CardText.tsx [hunk 4]
```diff
@@ -28,77 +28,75 @@ export interface CardTextProps {
...
 CardText.defaultProps = {
-  publisher: "Publisher",
-  author: "Author",
-  url: "https://www.test.com/",
-  title: "The title of this article is missing",
-  shortDesc: "This article does not have a description provided.",
+  publisher: 'Publisher',
+  author: 'Author',
+  url: 'https://www.test.com/',
+  headline: 'The title of this article is missing',
```
Comment (kkyeboah) [inline src/components/CardText/CardText.tsx]:
```suggestion
  title: 'The title of this article is missing',
```

File: src/components/CardText/CardText.tsx [hunk 5]
```diff
@@ -28,77 +28,75 @@ export interface CardTextProps {
...
   );
 };
 
 CardText.propTypes = {
   publisher: PropTypes.string.isRequired,
   author: PropTypes.string.isRequired,
   url: PropTypes.string.isRequired,
-  title: PropTypes.string.isRequired,
-  shortDesc: PropTypes.string.isRequired,
+  headline: PropTypes.string.isRequired,
```
Comment (kkyeboah) [inline src/components/CardText/CardText.tsx]:
```suggestion
  title: PropTypes.string.isRequired,
```

File: src/components/CardText/CardText.tsx [hunk 6]
```diff
@@ -28,77 +28,75 @@ export interface CardTextProps {
...
-          </a>
+    <CardContent className={classes.root}>
+      <Typography
+        className={classes.headline}
+        variant="h3"
+        align="left"
+        gutterBottom
+      >
+        <a className={classes.url} href={url}>
+          {headline}
```
Comment (kkyeboah) [inline src/components/CardText/CardText.tsx]:
```suggestion
          {title}
```

File: src/components/CardText/CardText.tsx [hunk 7]
```diff
@@ -28,77 +28,75 @@ export interface CardTextProps {
...
 /**
- * A card component that displays information about a news article.
+ * A card component that displays information about a story/news article.
  *
  * @param props: CardTextProps
  * @returns JSX.Element The rendered card
  */
 export const CardText = (props: CardTextProps): JSX.Element => {
-  const { publisher, author, url, title, shortDesc } = props;
+  const { publisher, author, url, headline, excerpt } = props;
```
Comment (kkyeboah) [inline src/components/CardText/CardText.tsx]:
```suggestion
  const { publisher, author, url, title, excerpt } = props;
```

File: src/components/CardText/CardText.tsx [hunk 8]
```diff
@@ -28,77 +28,75 @@ export interface CardTextProps {
...
-   * The URL of the article
+   * The URL of the story
    */
   url: string;
   /**
-   * The title of the article
+   * The title of the story
    */
-  title: string;
+  headline: string;
```
Comment (kkyeboah) [inline src/components/CardText/CardText.tsx]:
```suggestion
  title: string;
```

**Review by kkyeboah — CHANGES_REQUESTED** [review]
@nina-py Nice work adding the Edit and Approve component and cleaning up the other components. There's just a slight misalignment between the UI and how data is represented by the API.

I left comments where things need to be updated, I may have missed a few spots but I think you'll get the general direction. Let me know if you have any questions

File: src/components/EditAndApproveStory/EditAndApproveStory.tsx (hunk 1 above)
Comment (nina-py) [reply src/components/EditAndApproveStory/EditAndApproveStory.tsx]: All done @kkyeboah! I cleaned up all the fixup commits as well.

File: src/components/EditAndApproveStory/EditAndApproveStory.tsx [hunk 9]
```diff
@@ -0,0 +1,320 @@
...
+              }}
+              label="History"
+              fullWidth
+              multiline
+              rows={4}
+              variant="outlined"
+            />
+          </Grid>
+
+          <Grid item xs={12} className={classes.alignRight}>
```
Comment (kkyeboah) [inline src/components/EditAndApproveStory/EditAndApproveStory.tsx]: Let's get rid of this `Prioritize this story` toggle

**Review by kkyeboah — CHANGES_REQUESTED** [review]
I have one more thing for you to remove

File: src/components/EditAndApproveStory/EditAndApproveStory.tsx (hunk 9 above)
Comment (nina-py) [reply src/components/EditAndApproveStory/EditAndApproveStory.tsx]: Done! Do we need custom Switch styles or should these be removed, too?

Comment (kkyeboah) [reply src/components/EditAndApproveStory/EditAndApproveStory.tsx]: For now, we don't need the custom switch styles

Comment (nina-py) [reply src/components/EditAndApproveStory/EditAndApproveStory.tsx]: Removed these from the theme file :+1:

**Comment by kkyeboah** [comment]
@nina-py Why do we have all the components rendered in `App.tsx`, is that intended to stay or just for demo on the PR

**Comment by nina-py** [comment]
@kkyeboah, yes, that is just to demo the new form and all the other components that use the custom Button component. I can remove those now that you've reviewed everything?

**Comment by kkyeboah** [comment]
> @kkyeboah, yes, that is just to demo the new form and all the other components that use the custom Button component. I can remove those now that you've reviewed everything?

Cool, let's do that.

**Comment by nina-py** [comment]
> Cool, let's do that.

Done.

**Review by kkyeboah — APPROVED** [review]
Nice job 👍

**Comment by nina-py** [comment]
Thank you @kkyeboah!

**Pull request merged**